test/stake.test.ts

```diff
@@ -97,28 +111,129 @@ describe('StakingPool.stake', () => {
     });
   });
 
+  context("Many pool is opened", async () => {
+    const stakeAmount = utils.parseEther('100');
+
+    beforeEach('init the first round and time passes', async () => {
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).approve(testEnv.stakingPool.address, RAY);
+      await testEnv.stakingPool
+      .connect(deployer)
+      .initNewPool(rewardPersecond, startTimestamp_1, duration, testEnv.rewardAsset.address);
+
+
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).approve(testEnv.stakingPool.address, RAY);
+      const initTx = await testEnv.stakingPool
+      .connect(deployer)
+      .initNewPool(rewardPersecond, startTimestamp_2, duration, testEnv.rewardAsset.address);
+      await advanceTimeTo(await getTimestamp(initTx), startTimestamp_2);
+
+      await testEnv.stakingAsset.connect(alice).faucet();
+      await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+    });
+
+    it('staking pool 1, 2 are opened', async () => {
+      
+      //pool 1 staking
+      const poolDataBefore_1 = await getPoolData(testEnv, 1);
+      const userDataBefore_1 = await getUserData(testEnv, alice, 1);
+      const stakeTx_1 = await testEnv.stakingPool.connect(alice).stake(stakeAmount, 1);
+
+      const [expectedPoolData_1, expectedUserData_1] = expectDataAfterStake(
+        poolDataBefore_1,
+        userDataBefore_1,
+        await getTimestamp(stakeTx_1),
+        stakeAmount 
+      );
+
+      const poolDataAfter_1 = await getPoolData(testEnv, 1);
+      const userDataAfter_1 = await getUserData(testEnv, alice, 1);
+      
+      //pool 2 staking
+      const poolDataBefore_2 = await getPoolData(testEnv, 2);
+      const userDataBefore_2 = await getUserData(testEnv, alice, 2);
+      const stakeTx_2 = await testEnv.stakingPool.connect(alice).stake(stakeAmount, 2);
+
+
+      const [expectedPoolData_2, expectedUserData_2] = expectDataAfterStake(
+        poolDataBefore_2,
+        userDataBefore_2,
+        await getTimestamp(stakeTx_2),
+        stakeAmount
+      );
+
+      const poolDataAfter_2 = await getPoolData(testEnv, 2);
+      const userDataAfter_2 = await getUserData(testEnv, alice, 2);
+
+      //expect result
+      expect(poolDataAfter_1).to.be.equalPoolData(expectedPoolData_1);
+      expect(userDataAfter_1).to.be.equalUserData(expectedUserData_1);
+
+      expect(poolDataAfter_2).to.be.equalPoolData(expectedPoolData_2);
+      expect(userDataAfter_2).to.be.equalUserData(expectedUserData_2);
+            
+    });
+
+
+    it('staking pool 3 is opened, 1,2 are closed', async () => {
```

Comment:
The test description should be more explicit, explaining the action and expected result after the code is executed. 
How about 'it should be reverted when staking pool 3 is opened, 1,2 are closed.'